### PR TITLE
Move document information and outline to list view panel

### LINKED ISSUE #44193: Move document information and outline to list view panel
Consolidating list view / navigator, document stats, and document outline has surfaced a few times in the past. I'm proposing we revisit this and redesign the document stats a little bit. With the addition of "time to read" the current grid design is becoming harder to quickly parse:

- Display them as a list and smaller font size at the top or bottom of list panel.
- Add a tab interface to switch between all blocks and outline when headings are in place.

Mockup:

integrated-list-view

* List view and Outline tabs at the top
* Outline tab only available if the document has heading blocks
* Outline stats fixed at the bottom, shown on both tabs

Related #41330

Some questions to resolve in implementation: does the document stats take up valuable list view space? If yes, we can [collapse the information into a single row](https://github.com/WordPress/gutenberg/issues/44193#issuecomment-1252189792).

---

Issue updated Sep. 30.

### LINKED ISSUE #44862: The focus ring of document details is too big
### Description

In Firefox 105.0.3 the focus ring of document details is too big.

### Step-by-step reproduction instructions

1. In a post
2. Click on the (i) icon in the top toolbar
3. Notice the focus ring around the panel is the wrong size

### Screenshots, screen recording, code snippet

Screenshot 2022-10-11 at 11 18 16

### Environment info

WP 6.1
Gutenberg trunk
Firefox 105.0.3 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Fixes: https://github.com/WordPress/gutenberg/issues/44193
Fixes: #44862

Moves document information and outline to the list view panel on the edit post screen, trying to follow the mockups proposed in https://github.com/WordPress/gutenberg/issues/44193.

## Screenshots or screencast 

image
image

### CODE REVIEW

**Review by draganescu — COMMENTED** [review]
This appears to have a bug in Firefox:


https://user-images.githubusercontent.com/107534/195045611-d4cc0fdf-f2fd-4899-8bd6-17a728ec9d77.mp4

**Comment by jorgefilipecosta** [comment]
Nice catch @draganescu the issue was fixed 👍

File: packages/edit-post/src/components/secondary-sidebar/style.scss [hunk 1]
```diff
@@ -9,6 +9,10 @@
 	// Same width as the Inserter.
 	// @see packages/block-editor/src/components/inserter/style.scss
 	min-width: 350px;
+	& > .document-outline, & > .edit-post-editor__list-view-panel-content {
+		overflow-y: auto;
+		height: calc( 100% - 180px - #{$grid-unit-05});
```
Comment (draganescu) [inline packages/edit-post/src/components/secondary-sidebar/style.scss]: What is 180px?

File: packages/edit-post/src/components/secondary-sidebar/style.scss [hunk 2]
```diff
@@ -38,10 +42,39 @@
...
-	overflow-y: auto;
 	// The table cells use an extra pixels of space left and right. We compensate for that here.
 	padding: $grid-unit-10 ($grid-unit-10 - $border-width - $border-width);
 }
+
+.edit-post-editor__list-view-overview {
+	position: absolute;
+	bottom: 0;
+	& > div > span:first-child {
+		width: 90px;
```
Comment (draganescu) [inline packages/edit-post/src/components/secondary-sidebar/style.scss:81]: Same as with 180px, a comment on what it is if we cannot male it a var of sorts.

**Review by draganescu — COMMENTED** [review]
Lovely 👏🏻  This looks great now. Are these two tweaks worth a second look?

1. there is a jankiness when toggling inserter and list view
<details><summary>video</summary>


https://user-images.githubusercontent.com/107534/195593404-66be7293-7894-45aa-9e20-40d8593d9555.mp4



</details>

2. there is a page jump when toggling document structure and document outline
<details><summary>video</summary>


https://user-images.githubusercontent.com/107534/195593478-ed064832-dbe1-4dcb-ac1d-19798629053d.mp4



</details>

**Review by draganescu — DISMISSED** [review]
There is a missing feature from the original issue (#44193):

> Outline tab only available if the document has heading blocks

<details><summary>Example</summary>

Screenshot 2022-10-13 at 15 22 20


</details>

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 3]
```diff
@@ -43,10 +57,49 @@ export default function ListViewSidebar() {
 			onKeyDown={ closeOnEscape }
 		>
 			<div
-				className="edit-post-editor__list-view-panel-header"
+				className="edit-post-editor__list-view-panel-header components-panel__header edit-post-sidebar__panel-tabs"
 				ref={ headerFocusReturnRef }
 			>
-				<strong id={ labelId }>{ __( 'List View' ) }</strong>
```
Comment (alexstine) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:49]: This needs to stay for accessibility.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 4]
```diff
@@ -43,10 +57,49 @@ export default function ListViewSidebar() {
...
+				<ul>
```
Comment (alexstine) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:59]: This should appear after the Close button, not before for discovery reasons.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 5]
```diff
@@ -43,10 +57,49 @@ export default function ListViewSidebar() {
...
+						<Button
+							onClick={ () => {
+								setTab( 'outline' );
+							} }
+							className={ classnames(
+								'edit-post-sidebar__panel-tab',
+								{ 'is-active': tab === 'outline' }
+							) }
+							data-label={ __( 'Outline' ) }
+							aria-label={
```
Comment (alexstine) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Changing `aria-label` text based on selected state works but is not recommended. Please use `aria-current="true/false"` instead. You can add it the same way you add the class.

**Review by alexstine — COMMENTED** [review]
@jorgefilipecosta Thanks for the PR. This needs some accessibility work, please see comments below.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 5 above)
Comment (alexstine) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: The other option is to render a tab control here instead of buttons. That should work just as well if not better for this use-case.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 3 above)
Comment (jorgefilipecosta) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:49]: I added an aria-label instead aria-labelledby it should have the same effect.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 4 above)
Comment (jorgefilipecosta) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:59]: Applied the feedback 👍

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 5 above)
Comment (jorgefilipecosta) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Updated to use aria-current 👍

**Comment by jorgefilipecosta** [comment]
Hi @draganescu, @alexstine, thank you a lot for the reviews, I tried to apply all the feedback.

**Comment by alexstine** [comment]
@jorgefilipecosta Why does this info appear on List View and Outline tabs? Shouldn't it just appear on Outline tab?

> Characters:241
Words:42
Time to read:< 1 minute

This information is not expected in a List View tab especially since it appears after an application role.

**Comment by jorgefilipecosta** [comment]
> @jorgefilipecosta Why does this info appear on List View and Outline tabs? Shouldn't it just appear on Outline tab?
> 
> > Characters:241
> > Words:42
> > Time to read:< 1 minute
> 
> This information is not expected in a List View tab especially since it appears after an application role.

Hi @alexstine; thank you for letting me know about this issue.
The mockup shared by @mtias on https://github.com/WordPress/gutenberg/issues/44193 shows this information on the list view too. I'm following the proposed mockup.
Would making the application role include the list view blocks and the character/word count information be more accessible?
Could we keep character/word count information after the application role but add a different role to that information?
Any other ways that could make the interface accessible while complying with the design proposal?
Thank you in advance for any insights you may provide!

**Comment by alexstine** [comment]
@jorgefilipecosta Not really. You are dealing with a contextual issue. When the designers opened the door with tabs, you set the principal that there would be very defined content per tab. E.g. If you have a list view tab, this is okay, but that tab should only contain list view related content. The outline tab would make much more sense to find this information under and it would be more consistent with the idea of one defined item per tab.

Currently, user focus is placed in the application when the list view opens so users would have to know that other content is there to find it. You cannot add this info to the application because this is what makes the tree grid items accessible. Adding another role could actually make things inaccessible because of the very nature of how screen readers are choosing to work now.

TLDR: No.

Thanks.

**Comment by jorgefilipecosta** [comment]
@mtias, @jasmussen, @jameskoster given the acessibiity contraints referred above would it be ok to have character word/count just on the outline tab?

**Comment by jameskoster** [comment]
That seems reasonable to me, indeed we discussed on the original ticket. 

It does mean we'll need to display the Overview tab permanently though (so stats are always accessible). Consequently the Overview tab will need an 'empty' design... for now it can probably just contain a description.

**Comment by jorgefilipecosta** [comment]
Thank you for the insights @jameskoster! This PR was updated. The character/word count just appears on the outline, we have an empty message for the case where there are no headings.

**Comment by jameskoster** [comment]
Looking good @jorgefilipecosta. Could we use the same tab style as the Inserter with the left-justified text?

A minor point, but perhaps instead of the generic "There are no headings on this post." message, we might offer an explanation of the feature, something like:

Screenshot 2022-10-20 at 15 46 38

cc @WordPress/gutenberg-design for any thoughts on that.

**Comment by javierarce** [comment]
@jameskoster I think it's a great idea. I would only change the illustration a bit to match the actual style of the outlines and simplify the copy to read: "Navigate the structure of your document  and address issues like empty or incorrect heading levels."

image

[Figma file](https://www.figma.com/file/bqdsT0AetJWnETqD1h7XWX/%E2%9B%85%EF%B8%8FExplorations?node-id=1416%3A50373)

**Review by alexstine — COMMENTED** [review]
The code and functionality work for me. However, some part of me says this feels wrong having not extracted some of the extra functionality from the list view sidebar. We have a `ListView` component, why not create a `ListViewOutline` or similar to keep this file a little cleaner?

**Comment by jorgefilipecosta** [comment]
> The code and functionality work for me. However, some part of me says this feels wrong having not extracted some of the extra functionality from the list view sidebar. We have a `ListView` component, why not create a `ListViewOutline` or similar to keep this file a little cleaner?

Thank @alexstine your feedback was applied.

**Comment by jorgefilipecosta** [comment]
Thank you @javierarce, and @jameskoster for providing this design and the Figma file. The Figma files make everything much easier. The design was passed to code.

File: packages/edit-post/src/components/secondary-sidebar/empty-outline-ilustration.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { Path, SVG, Line, Rect } from '@wordpress/components';
+
+export default function EmptyOutlineIlutration() {
```
Comment (alexstine) [inline packages/edit-post/src/components/secondary-sidebar/empty-outline-ilustration.js]: Is that a spelling error? Probably should fix and fix the references.

```suggestion
export default function EmptyOutlineIllustration() {
```

**Review by alexstine — COMMENTED** [review]
@jorgefilipecosta I would probably combine these files.
- packages/edit-post/src/components/secondary-sidebar/empty-outline-ilustration.js
- packages/edit-post/src/components/secondary-sidebar/list-view-outline.js

You can keep the same code but just combine them since they are pretty much the same concept. I might also move those to packages/block-editor/src/conponents/list-view just to keep everything together or at the very least packages/block-editor/src/components/outline-tab or similar.

@talldan Do you have any feedback as to how to create a better component structure here? I have done mostly modifications to React, nothing new.

**Comment by talldan** [comment]
The structure looks ok to me.

I noticed there's quite a lot of inconsistency with the styling of tabs in the block editor now though, not just the alignment and sizing of the tabs, but there are a few subtle differences in height too (cc @jameskoster), might be a good opportunity to tidy this up:
Screen Shot 2022-10-24 at 2 48 06 pm
Screen Shot 2022-10-24 at 2 47 58 pm
Screen Shot 2022-10-24 at 2 47 49 pm

The last one is probably best for i18n, as there's more flexibility in text width.

When creating a new post List View can also be empty, so some placeholder state to match the outline's might be good there too, but it could always be a separate issue (a good one for a first time contributor.) 👍

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 4 above)
Comment (talldan) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:59]: This does make the tab order not match the visual order, and there's also now an inconsistency with the settings sidebar and how its tabs and close button are ordered.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 7]
```diff
@@ -32,35 +38,68 @@ export default function ListViewSidebar() {
...
+							className={ classnames(
+								'edit-post-sidebar__panel-tab',
+								{ 'is-active': tab === 'outline' }
+							) }
+							aria-current={ tab === 'outline' }
+						>
+							{ __( 'Outline' ) }
+						</Button>
+					</li>
+				</ul>
```
Comment (talldan) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:88]: This should probably use the `TabPanel` component - https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/tab-panel/README.md

**Comment by jameskoster** [comment]
> might be a good opportunity to tidy this up

I think aligning this panel with the Inserter (in terms of tab style) is ok for now. Long term we have https://github.com/WordPress/gutenberg/issues/42320 and https://github.com/WordPress/gutenberg/issues/41904 to address the issues you mention.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 7 above)
Comment (jorgefilipecosta) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:88]: The TabPanel component does not support a close button at the side of the tabs. That's the reason we are not using it. The same happens with the main tabs between the block and the document. I'm following the approach used there for consistency.

File: packages/edit-post/src/components/secondary-sidebar/empty-outline-ilustration.js (hunk 6 above)
Comment (jorgefilipecosta) [reply packages/edit-post/src/components/secondary-sidebar/empty-outline-ilustration.js]: Thank you, the typo was fixed 👍

**Comment by jorgefilipecosta** [comment]
Hi @alexstine,

> @jorgefilipecosta I would probably combine these files.
> 
> * packages/edit-post/src/components/secondary-sidebar/empty-outline-ilustration.js
> * packages/edit-post/src/components/secondary-sidebar/list-view-outline.js
> 
> You can keep the same code but just combine them since they are pretty much the same concept.

The feedback was applied.


> I might also move those to packages/block-editor/src/conponents/list-view just to keep everything together or at the very least packages/block-editor/src/components/outline-tab or similar.

We can not move the outline to the block-editor package because it depends on the editor package that is WordPress specific. And the block-editor should not be WordPress specific.

**Comment by jorgefilipecosta** [comment]
> > might be a good opportunity to tidy this up
> 
> I think aligning this panel with the Inserter (in terms of tab style) is ok for now. Long term we have #42320 and #41904 to address the issues you mention.

Hi @jameskoster, I updated the design to be similar to the inserter.

I think there is no pending feedback, and this PR is ready.

**Review by alexstine — APPROVED** [review]
LGTM.

**Comment by talldan** [comment]
> The TabPanel component does not support a close button at the side of the tabs.

@jorgefilipecosta Isn't that something to try fixing? With this PR there will be four separate tab implementations (possibly even more) and they're all visually similar to `TabPanel`, but with lots of inconsistencies.

Three of them have a close button, so that pattern is well established and could be supported by the `TabPanel` component.

**Comment by jorgefilipecosta** [comment]
> Three of them have a close button, so that pattern is well established and could be supported by the TabPanel component.

I'm not sure if the TabPanel should have a close button or not, as it is not really very expected to have a close button on tabs for the most common tab uses. Maybe @ciampo has some thoughts on this.
I guess if we want TabPanels there we can try to move all the tabs to that component.

**Comment by talldan** [comment]
Probably a few different ways to do it, but there is an option of keeping the close button and tab panel as separate components and making them part of a horizontal layout. The main challenge seems to be the bottom border that tab panel shows.

**Comment by ciampo** [comment]
> Isn't that something to try fixing? With this PR there will be four separate tab implementations (possibly even more) and they're all visually similar to `TabPanel`, but with lots of inconsistencies.

I couldn't agree more with @talldan 's point above about not having 4+ slightly different implementations of `TabPanel`.

Whenever a component doesn't satisfy certain requirements, we should try to work and improve it to satisfy those requirements and/or tweak them if needed. Avoiding the issue by creating a new component introduces fragmentation, technical debt and ultimately hurts the project in the long term.

> I'm not sure if the TabPanel should have a close button or not, as it is not really very expected to have a close button on tabs for the most common tab uses. Maybe @ciampo has some thoughts on this.

I don't think that the close button should be part of `TabPanel`. The close button closes the sidebar, while `TabPanel` takes care of whatever content is _inside_ that sidebar.

Another thing to take into consideration is that the monolithic nature of `TabPanel` doesn't really allow to inject any component to the side of the tabs anyway.

> there is an option of keeping the close button and tab panel as separate components and making them part of a horizontal layout. The main challenge seems to be the bottom border that tab panel shows.

This sounds like a good direction to explore:

- the close button could be `absolute` position (`top: 0; right: 0`)
- we could add some right padding by targeting in the tabs container with the CSS selector `.my-tabpanel-custom-classname [role="tablist"]`
- regarding the light gray border between the tablist and each tab panel, this is probably something that needs to be codified in our design system first, so that we ensure some coherence across the editor's UI. Should our `TabPanel` component always show a border under the tablist, never show it, or show it under specific circumstances?

**Comment by alexstine** [comment]
A few notes:
1. I agree with using `TabPanel` here and we should really use it globally around the editor.
2. This did follow perfect accessibility, exact same functionality for the main sidebar.
3. The `TabPanel` component needs a modification to allow passing a prop for receiving enter/space press. Every time the `ListView` mounts, focus is placed on it. However, this occurs on a button click at the moment. Changing to `TabPanel` would make it so the left arrow focus on List View item would focus the `ListView`. If we instead were able to pass a prop to the `TabPanel` component that waited for space/enter key to select the tab, that would help us here and future PRs such as https://github.com/WordPress/gutenberg/pull/45203.

My main problem is, no one ever has time to work on these small enhancements because the project moves too quickly so sometimes I settle for less for reasons only related to accessibility. At this point, I would call `TabPanel` bad UX until my suggestion is implemented for this use-case.

Thanks.

**Comment by ciampo** [comment]
Hey @alexstine !

> This did follow perfect accessibility, exact same functionality for the main sidebar.

I'm not not sure I understand what you meant here, would you mind clarifying?

> Every time the ListView mounts, focus is placed on it. However, this occurs on a button click at the moment. Changing to `TabPanel` would make it so the left arrow focus on List View item would focus the `ListView`

Again, I'm not sure I understand the issue that you're pointing out. Looking at [the `TabPanel` component](https://wordpress.github.io/gutenberg/?path=/story/components-tabpanel--default), it is possible to move focus (via the `tab` key) on the tablist element and then press the left/right arrow keys to select the different tabs, which in turn automatically selects and shows the associated tab panel. This is referred to as "Tabs with automatic activation" on [the W3C website](https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/).

> If we instead were able to pass a prop to the TabPanel component that waited for space/enter key to select the tab, that would help us here and future PRs such as https://github.com/WordPress/gutenberg/pull/45203.

Using space/enter is indeed another variant of the Tabs pattern, described as "Tabs with manual activation" on the same [W3C Website](https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/), which currently the `TabPanel` component doesn't support.

What would be the exact reason for using manual activation over automatic activation?

Once these requirements are a bit more clear, we may open a PR and discuss the next steps for the `TabPanel` component.

> My main problem is, no one ever has time to work on these small enhancements because the project moves too quickly so sometimes I settle for less for reasons only related to accessibility. At this point, I would call TabPanel bad UX until my suggestion is implemented for this use-case.

While it is true that a good part of the development work in Gutenberg goes towards adding new features and evolving the product, I can guarantee you that at the same time we try to do our best work in constantly polishing and improving our code. Folks like @mirka and myself, for example, spend most of their time improving the `@wordpress/component` package — which usually means fixing bugs (including accessibility ones) and reducing technical debt. We always really appreciate and value your feedback, and we try to apply it where needed as much as possible.

Finally, I'm not sure I'd call the `TabPanel` component "bad UX", as it seems to largely follow the W3C ARIA specs for the Tabs pattern — but at the same time, we're definitely open to feedback and always look for ways to improve our components.

**Comment by alexstine** [comment]
@ciampo Okay, I guess I should have had my coffee before writing this. Let me explain.

> I'm not not sure I understand what you meant here, would you mind clarifying?

As far as the List View and Outline tabs, they are the same as the document settings/block settings sidebar. Those are perfectly accessible, that is why the PR was okay to me. It follows a model already used in the editors.

> Again, I'm not sure I understand the issue that you're pointing out. Looking at the TabPanel component, it is possible to move focus (via the tab key) on the tablist element and then press the left/right arrow keys to select the different tabs, which in turn automatically selects and shows the associated tab panel. This is referred to as "Tabs with automatic activation" on the W3C website.

We cannot use automatic activation here because the `ListView` component mounts and fires the `useFocusOnMount` hook. This would introduce an unexpected focus jump if simply exploring the tabs. Switching to manual activation would be better in this case that way the `ListView` mounts after the user manually selects the tab with space/enter key. Does this make sense? Automatic activation is fine but when automatic focus is at play, it is not a great situation because the user could simply be exploring their options and trigger focus by mistake.

As far as the mentioned prop, this would not need to be changed globally. This is a case-by-case thing. That is why I suggested adding a new prop to `TabPanel` controlling this via true/false defaulting to false.

> Finally, I'm not sure I'd call the TabPanel component "bad UX", as it seems to largely follow the W3C ARIA specs for the Tabs pattern — but at the same time, we're definitely open to feedback and always look for ways to improve our components.

The `TabPanel` is not bad UX, automatic focus changes would be bad UX as it relates to this specific PR and the related PR I mentioned above. The `TabPanel` is great, it just does not currently meet our use-case in these 2 PRs. If we can modify it, that would be amazing. Remember, the golden rule is only move focus if the user is aware. There is no way a user could be aware that changing tabs will trigger a focus jump. That is why space/enter keys would be necessary to use `TabPanel` for these 2 PRs which would make it great UX. We're currently faking this with buttons right now but it would be great to move away from that globally.

I know you work hard to improve but at the same time, know that the vast majority of PRs I review for A11Y are new features/changing UIs. There should be a larger focus on improving what we already have, that was my point for the project as a whole, not specific to your great work on components. What good are accessible components if they are used in inaccessible contexts. That is what I am fighting to improve.

Sorry my above comment came off so rough but I meant nothing against the contributors over the components team. That feedback was for this PR alone.

Thanks.

**Comment by ciampo** [comment]
Thank you for the context and the detailed explanation, @alexstine !

I went ahead and opened a new issue related to adding support for manual tab activation to `TabPanel`: https://github.com/WordPress/gutenberg/issues/45390


> Sorry my above comment came off so rough but I meant nothing against the contributors over the components team. That feedback was for this PR alone.

Don't worry about it! As I mentioned before, we _always_ appreciate your contributions and your will to improve the accessibility of the editor.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 4 above)
Comment (afercia) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js:59]: > This does make the tab order not match the visual order

Yep, I just noticed this and it's a valid concern. I'll create a new issue to report other a11y improvement as well.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 5 above)
Comment (afercia) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: As @alexstine suggested, an ARIA tabbed interfaced would be better. Will report in a new issue. Also, the initial focus skips the 'tabs'.

Comment (ciampo) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: > Will report in a new issue.

When doing so, could you mention https://github.com/WordPress/gutenberg/issues/45390 being a dependency? From what I understand, until manual tab activation is added to `TabPanel`, we won't be able to use it here

Comment (afercia) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Sure, thanks. If the problem is the initial focus, I think we should move `focusOnMountRef` to the `ListViewSidebar` container rather than set it on the list view container. That would avoid the focus to skip the tabs, regardless whether they're manually or automatically activated.

Comment (alexstine) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: @afercia That is not an option because the `ListView()` is rendered as an `role="application"`. Focus needs to be placed there on mount even if we risk skipping the tabs. There is a long outstanding NVDA bug about this.

https://github.com/nvaccess/nvda/issues/12896

Comment (afercia) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Thanks @alexstine can you please mention this on the new issue #45899 ?
Will read about the NVDA bug as soon as I have some time.
